core/embed/rust/src/ui/component/marquee.rs

```diff
@@ -0,0 +1,229 @@
+use crate::{
+    micropython::buffer::StrBuffer,
```

Comment:
agree, though here is my issue: this is used in TR branch from `Title` component (https://github.com/trezor/trezor-firmware/blob/grdddj/model_r_smaller/core/embed/rust/src/ui/model_tr/component/title.rs) , which also uses `StrBuffer`. So we would need to either:
- change `Title` (and components using it) not to use `StrBuffer`
- use `as_ref` when creating `Marquee` and add lifetimes to `Marquee` (and components that use it)
- possibly change `as_ref` of StrBuffer to return `&'static str`. Not sure whether this would be possible and correct.
- something else? any ideas?

There is additional related problem, many of the TR components use `StrBuffer`, but i need to use them from bootloader without micropython. So what i did in the TR bootloader PR is to actually move the `StrBuffer` out of micropython module and compile `StrBuffer` with or without micropython related stuff based on `#[cfg(feature = "micropython")]`. This would solve the problem here too. See here: https://github.com/trezor/trezor-firmware/blob/tychovrahe/rust_bootloader_/tr/core/embed/rust/src/strbuffer.rs (note that this PR is still very draftish, maybe it would be nicer not to duplicate the file and just conditionally compile specific functions or something)

Reply:
Hmm also making Marquee generic would work and seems most painless

Reply:
did the generic approach: https://github.com/trezor/trezor-firmware/commit/a62e5de96358f93f2acdf409c55fbc3fe9878005

Reply:
yeah, making the str type generic is ugly but it's the simplest solution